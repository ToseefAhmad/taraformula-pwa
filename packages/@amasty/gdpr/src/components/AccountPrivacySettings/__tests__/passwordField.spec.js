import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import PasswordField from '../passwordField';

jest.mock('@magento/venia-ui/lib/components/Password', () => 'Password');
jest.mock('react-intl', () => ({
  useIntl: jest.fn().mockReturnValue({
    formatMessage: jest
      .fn()
      .mockImplementation(options => options.defaultMessage)
  })
}));

test('renders needPassword false', () => {
  const props = {
    item: {
      needPassword: false,
      action: true
    },
    isShowPassword: true
  };

  const tree = createTestInstance(<PasswordField {...props} />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('renders action false', () => {
  const props = {
    item: {
      needPassword: true,
      action: false
    },
    isShowPassword: true
  };

  const tree = createTestInstance(<PasswordField {...props} />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('renders isShowPassword false', () => {
  const props = {
    item: {
      needPassword: true,
      action: true
    },
    isShowPassword: false
  };

  const tree = createTestInstance(<PasswordField {...props} />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('renders default', () => {
  const props = {
    item: {
      needPassword: true,
      action: true
    },
    isShowPassword: true
  };
  const tree = createTestInstance(<PasswordField {...props} />);
  expect(tree.toJSON()).toMatchSnapshot();
});
