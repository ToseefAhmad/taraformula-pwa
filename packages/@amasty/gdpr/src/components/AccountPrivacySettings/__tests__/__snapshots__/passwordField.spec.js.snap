// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders action false 1`] = `null`;

exports[`renders default 1`] = `
<Password
  classes={
    Object {
      "root": undefined,
    }
  }
  fieldName="password"
  isToggleButtonHidden={false}
  label="Current Password"
  validate={[Function]}
/>
`;

exports[`renders isShowPassword false 1`] = `null`;

exports[`renders needPassword false 1`] = `null`;
